passes/opt/opt_expr.cc

```diff
@@ -604,7 +604,7 @@ void replace_const_cells(RTLIL::Design *design, RTLIL::Module *module, bool cons
 				if (cell->type.in(ID($xnor), ID($_XNOR_))) {
 					cover("opt.opt_expr.const_xnor");
 					// For consistency since simplemap does $xnor -> $_XOR_ + $_NOT_
-					int width = cell->getParam(ID::Y_WIDTH).as_int();
+					int width = GetSize(cell->getPort(ID::Y));
```

Comment:
Wait, why does this make a difference? Shouldn't a module where these two don't match fail `check()`?

Reply:
They always match for `$xnor`, but `Y_WIDTH` is not present at all for `$_XNOR_`, causing the crash reported in #2311.